### ISSUE TITLE
[Question] Host a authenticated NuGet v3 feed?

### ISSUE THREAD

**Issue by DoCode** [emgarten/Sleet#36]
Is it possible to host an authenticated (basic auth), NuGet v3 feed, created with the Sleet toolchain, that Visual Studio PM, NuGet CLI, and DotNet CLI can consume packages?

**Comment by emgarten** [comment]
For an authenticated feed we need either the NuGet client to support azure storage auth, or to put something in front of storage to auth.

The easiest way to do this would be to use azure functions to serve up files from the storage account, and auth with basic auth. In the sleet config you can set `baseURI` to give it a different URI than the direct storage account.

Client support is tracked here: https://github.com/NuGet/Home/issues/6574

I would really like to have an example or azure function code that could be used for this also, I think support for private feeds is a primary piece that is needed right now.

**Comment by DoCode** [comment]
Thanks!
I think a very simple solution, i.e. a dotnet core frontend should provided. Because the S3 or filesystem storage provider has the same requirements.

**Comment by emgarten** [comment]
That's true, it could use the same filesystem. So you would be hosting this on AWS?

**Comment by DoCode** [comment]
Currently, we will it be hosting on a local S3 storage system (minio.io).
But I think a default frontend (.NET Core) proxy would be a generic solution, for any backend (filesystem, Azure, and S3).
What do you think?

**Comment by emgarten** [comment]
Ideally what would the proxy component look like for you? A docker container that runs along side minio.io? A local app that runs a proxy on your dev box to talk to the private feed's file system?

**Comment by DoCode** [comment]
Yes, that's my opinion. A very simple proxy that could host everywhere (docker, Azure, AWS, IIS) and which have access to the `backend` sleet package store, like the file system, Azure or S3.

**Comment by gyamada619** [comment]
Any chance adding basic auth support for the client trying to access S3 is closer to being supported? We're looking for exactly this -- a NuGet feed in S3, which we can in theory store our Chocolately packages on.

**Comment by emgarten** [comment]
To add auth on s3 you would need to serve up the json content through a lambda or a different endpoint that can handle auth.

Possibly some very hacky auth could be done by creating a feed on s3 using `baseURI` to set a different URI in the json files. Then serve this up with CloudFront and add a rule there to check for the expected basic auth header.

I know this is a common scenario, so I'm open to suggestions on what might be a good way to add auth support for s3 and azure.

**Comment by davidturvey** [comment]
I have found a possible solution/workaround for AWS. It is possible to add basic auth to an s3 bucket using a lambda@edge function and cloudfront. I followed this guide ( http://kynatro.com/blog/2018/01/03/a-step-by-step-guide-to-creating-a-password-protected-s3-bucket/ ) and it seems to be working well so far. 

The only change i made to the sleet config was to set the cloudfront address as the baseURI.
